prime-router/metadata/HL7/catchall/hl7/datatypes/XAD/Address.yml

```diff
@@ -38,7 +38,7 @@ postalCode:
 country:
   type: STRING
   expressionType: HL7Spec
-  valueOf: XAD.6
+  valueOf: $country | XAD.6
```

Comment:
Where is $country coming from?

Reply:
Looks like this was a miss by me. Looking at the v251 mappings we are using it in Organization.address and I didn't add that in.

Reply:
TLDR: [I updated HD/Organization to use datatypes/XAD/Address instead of datatype/Address](https://github.com/CDCgov/prime-reportstream/pull/15333/commits/4e1f1df59e7d8b7f1571a341e2971d7023c1f816). No tests failed and no tests had to be added because I already identified this as a testing gap [earlier on this PR](https://github.com/CDCgov/prime-reportstream/pull/15333/commits/5d2ba2a37eac3c96b21f614885f9ddecb29b8df9). (As extra confirmation I re-ran the fhirdata script on this directory and nothing other than ids were updated). @brick-green or @mkalish can you confirm that this is the correct course to take?

I'm going down a rabbit hole of "why is this thing the way it is" questions and will document that here to see if maybe somebody can sort me out, or at least leave some notes.

The original miss that I had here was adding in `$country` to XAD/Address without having it be used anywhere.

At first I thought it should be easy to parse out how this is being used in the v251 mappings and apply that here, but:
- HD/Organization is the only place using `$country` for XAD/Address in v251. HD/Organization is using datatype/Address in the catchall mappings.
- The catchall mappings have two Address.yml: datatype/Address.yml and datatypes/XAD/Address.yml. (v251 only appears to contain XAD/Address.yml).
  - Both of these are invoked in various catchall mapping files
  - Primarily datatype/Address is used by files in the secondary and resource directories that I have nominated for deletion. (https://github.com/CDCgov/prime-reportstream/issues/15342)
  - The only other use of datatype/Address is here in HD/Organization.

So what I imagine happened is that datatype/Address was used when the catchall were created because that had the country variable already defined and datatypes/XAD/Address did not. 

HD/Organization is the only place I could find the $country var being used, and this gets pulled in from MSH.17. (It took me awhile to realize that this mapping was documented on [MSH[MessageHeader]](https://docs.google.com/spreadsheets/d/13pgda5xl-PwCgB9j0axyymwwv7RJVcrIzY8Ah1y1Y1M/edit?gid=0#gid=0) and not HD[Organization].

Reply:
Huddled with Brick and this was approved.

Reply:
Due to changes in how XAD.6 is handled (see this comment: https://github.com/CDCgov/prime-reportstream/pull/15333#discussion_r1702332661) $country is no longer needed for XAD.6. See the comment for related commits removing it.